snapcraft/plugins/v1/catkin.py

```diff
@@ -321,6 +334,12 @@ def __init__(self, name, options, project):
         self.catkin_packages = None
         if options.catkin_packages is not None:
             self.catkin_packages = set(options.catkin_packages)
+
+        # Get a unique set of skip-keys
+        self.skip_keys = None
```

Comment:
If this was always a set it would simplify your later logic. Note that `None` is a special value for `cakin_packages`, so that's a part of the pattern we don't need here.

```suggestion
        self.skip_keys = set()
```